### PR TITLE
Add TShockPrometheus

### PR DESCRIPTION
**Pull request by swantzter**
Admittedly this is in somewhat of a proof-of-concept stage at the moment, but I intend to build it out with more metrics and hope that submitting it here could lead to more eyes on it to get suggestions for interesting metrics to add.

First proposed [here](https://github.com/Pryaxis/TShock/discussions/2196)

### CODE REVIEW

**Comment by bartico6** [comment]
Plugin idea itself is solid, looks good to me.

Metrics you could add, off the top of my head:

- Average tickrate over three polling periods (1m 5m 15m? like Spigot/Bukkit)
- Uptime
- Player count
- Average frame time (this is different than tickrate! Measure the time between each game-wide update call start & end for this)
- Memory usage?

File: README.md [hunk 1]
```diff
@@ -193,6 +193,11 @@ This is the TShock plugin repository. It collects links to all plugins in the ec
    * Tested on TShock 4.4.0 Pre-12.
    * [Download Version 1.0.0](https://files.catbox.moe/1ouznb.dll)
    * [Documentation and SourceCode](https://github.com/dredhorse/tShockAutoTeam)
+* [TShockPrometheus](https://github.com/swantzter/tshock-prometheus) by [Swantzter](https://github.com/swantzter)
+  * Exposes statistics about the server to prometheus
```
Comment (hakusaro) [inline README.md:197]:
```suggestion
  * Exposes the current player count on the server to prometheus
```

**Review by hakusaro — CHANGES_REQUESTED** [review]
Can you provide a link to `System.ValueTuple.dll` and `Prometheus.NetStandard.dll` from somewhere on the Internet (nuget?) before approval?

I've also made a suggestion about the description -- until you add more metrics, you should be pretty specific in what the plugin does.

Once you provide the original links to the DLLs above for analysis, I can continue the review! Thank you for your hard work on this!

Screen Shot 2021-01-03 at 11 27 11

**Comment by swantzter** [comment]
@hakusaro `Prometheus.NetStandard.dll` and `System.ValueTuple.dll` comes from installing https://www.nuget.org/packages/prometheus-net/ source on https://github.com/prometheus-net/prometheus-net (the library [listed by prometheus itself](https://prometheus.io/docs/instrumenting/clientlibs/)) (I assume it pulls `System.ValueTuple` from https://www.nuget.org/packages/System.ValueTuple/ I'm not too familiar with .NET buildchains)

**Comment by swantzter** [comment]
@bartico6 Oh yeah those are some great ideas, thanks! I had tickrate and total player count among my ideas already, interesting that frame rate is different!
Memory usage is apparently measured among the prometheus library's default metrics, but I'm planning to redo that a bit to give the metric name a proper `tshock_` prefix.

Currently trying to dig around my IntelliSense to figure out how to best dig out these various data points 😅 I'm not a C# developer so I'm admittedly a bit lost

**Review by hakusaro — APPROVED** [review]
lgtm_6

**Review by hakusaro — APPROVED** [review]

**Pull request merged**